rtlib/rope.hh

```diff
@@ -405,8 +435,57 @@ class Ref {
         append(s);
     }
 
-    template <typename O>
-    void put(O &o) const {
+    void put(std::ofstream &o) const {
+      if (readonly == true) {
```

Comment:
from just reading the code, it is not clear to me what the difference between readonly true and false are. Especially, since the nested for loop looks relatively similar. Can you leave a comment about it here?

Reply:
I'm not sure either, but I also didn't modify any of this. It looks like I did in the diff view, but I only changed the method signature. I left the implementation as it was before.